.travis.yml

```diff
@@ -379,6 +379,12 @@ before_script:
 
 # ----------------------------------- Jobs -----------------------------------
 
+stages:
+  - compile
+  - test
```

Comment:
What's the separation of these two stages for? Remember #396, I had seen tangible issues with caching between two jobs of *the same* build... Even if these are the same build chain, the two jobs are entirely separate virtual machines running, and it seemed that -- for the dependencies, as #396 was meant to cache that -- there is some weird nondeterminism...

Reply:
The default stages in Travis are: `compile`, `test` and `deploy`. Until now we have not used stages and all jobs were put into the `test` stage by default.

Stages are sequential, so e.g. the jobs in the `deploy` stage are only executed when all jobs in the previous `test` are finished (and passed). That was my aim here: to only build the Docker images and publish them if the previous jobs have succeeded, because otherwise something might be off and we shall not roll out the new Docker images to the public.
This deploy stage in our case is also configured that it is only executed in the original repository (not in the forked ones) and only upon pushes to the `master` branch (pull requests to the `master` are also excluded that way).

The `compile` stage can be omitted and removed, because we do not use it. Initially I have just listed the default stages and it was left there.

Reply:
Alright. That distinction is okay with me. I just wanted to get ahead of the possibility of trying to *share* things between the stages, as I have experienced it not working in the recent past. (We can't really split the "compile" and "test" stages without sharing the build between the two...)

Reply:
Yes, Travis has only a limited support to transfer, well, *anything* between stages. Only solvable with external providers, e.g. with Amazon S3.
https://docs.travis-ci.com/user/build-stages/#data-persistence-between-stages-and-jobs
https://docs.travis-ci.com/user/build-stages/share-files-s3/

Fortunately we do not need this functionality now, the Docker images are built from scratch each time.
(GitLab e.g. performs way better in this aspect, supporting to transfer so called *artifacts* between jobs and stages.)

Reply:
So, should we remove "compile" from the list, and just keep "test"?

Reply:
I have removed it now to avoid any possible confusion.